### PR TITLE
Add PythonOperator

### LINKED ISSUE #12: Python Operator
#8 only implements the `BashOperator`, but we will at the very least require a `PythonOperator` to call python functions.
It should:
- check if the function exists
- allow for aliasing of name spaces (e.g. "aliases: {'numpy': 'np'}" in the YAML file
- ability to inspect the function for input and output types (important for provenance and debugging)

### PR DESCRIPTION
**Pull request by kreczko**
Fixes #12 

- add `fasthep_flow.operators.PythonOperator`
- add `ResultType` for validating output

`ResultType` currently does nothing, need to see how chaining functions will interact here before further implementation

### CODE REVIEW

**Comment by YeungOnion** [comment]
Rust has a pattern for handling results that are mutually exclusive success or failure, stderr poses some distinction here.

I got stuck in the, "how do I do this in a way that handles module aliases?" and then never got back to it. Can I ask how you test the example? I'm expecting to test with `hatch`

**Comment by kreczko** [comment]
> how do I do this in a way that handles module aliases?
Also got stuck here, not quite sure atm. Currently trying some things to get `numpy` work as `np`.

> Can I ask how you test the example?
Currently running
```
fasthep-flow execute docs/examples/hello_world.yaml
```
in a conda env (python 3.12).
As this projects matures I will add tests to run with
```
pipx run nox -s tests
```
including the examples (hello world + CMS open data)

The result dict will likely change as I am looking at https://hamilton.dagworks.io/en/latest/reference/result-builders/Generic/ and the other variants. I would like something that is not dependent on hamilton, but can pass on enough metadata to convert correctly if hamilton is picked. Since awkward array has integrations with Dask, a custom ResultBuilder would be needed at some point.

Long story short, this project is still in alpha - trying to find its sweet-spot for implementation. Expect occasionally some incomplete functionality.
That said, I finally have some time to push towards a beta that matches the current fast-flow release in terms of features.
